packages/react/src/components/AccountSettings/ConfigureTOTP/ConfigureTOTP.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import QRCode from 'qrcode';
+
+import {
+  AmplifyUser,
+  getLogger,
+  getTotpCodeURL,
+  setupTOTP,
+  translate,
+  verifyTOTPToken,
+} from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import { FormValues } from '../types';
+import {
+  ConfirmationCode,
+  CopySecretKey,
+  Error,
+  SecretKeyQRCode,
+  SubmitButton,
+} from './defaults';
+import { ConfigureTOTPProps } from './types';
+
+const logger = getLogger('Auth');
+
+function ConfigureTOTP({
+  totpIssuer = 'AWSCognito',
+  totpUsername,
+  onSuccess,
+  onError,
+}: ConfigureTOTPProps): JSX.Element | null {
+  const [secretKey, setSecretKey] = React.useState<string>(null);
+  const [qrCode, setQrCode] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({ code: '' });
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  const { user, isLoading } = useAuth();
+
+  const hasInit = React.useRef(false);
+
+  const generateQRCode = React.useCallback(
+    async (user: AmplifyUser): Promise<void> => {
+      try {
+        const newSecretKey = await setupTOTP(user);
+        const username = totpUsername || user?.username;
+        const totpCode = getTotpCodeURL(totpIssuer, username, newSecretKey);
+        const qrCodeImageSource = await QRCode.toDataURL(totpCode);
+
+        if (!hasInit.current) {
+          setSecretKey(newSecretKey);
+          setQrCode(qrCodeImageSource);
+          hasInit.current = false;
+        }
+      } catch (e) {
+        logger.error(e);
+      }
+    },
+    [totpIssuer, totpUsername]
+  );
+
+  React.useEffect(() => {
+    if (user && !hasInit.current) {
```

Comment:
Is there a scenario where `user` is falsy?

Reply:
No, it is an assumption that `user` is already authenticated by the time customer runs this component. 

If not, `useAuth` and early return by logging error: https://github.com/aws-amplify/amplify-ui/blob/a62a1f08d7eaad99ce6909dc0ae72eb0939ffdb1/packages/react/src/components/AccountSettings/ConfigureTOTP/ConfigureTOTP.tsx#L117-L120